src/test/java/TicTacToe/ComputerPlayerTest.java

```diff
@@ -0,0 +1,86 @@
+package TicTacToe;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ComputerPlayerTest {
+    @Mock
+    Board board;
+    @Mock
+    RandomWrapper randomWrapper;
+    ComputerPlayer computerPlayer;
+    private static final char CIRCLE = Figures.CIRCLE.getCharacter();
+    private static final char CROSS = Figures.CROSS.getCharacter();
+    private static final char EMPTY = Figures.EMPTY.getCharacter();
+
+    @BeforeEach
+    public void mockClass() {
+        computerPlayer = new ComputerPlayer(board, randomWrapper);
+    }
+
+    @Test
+    public void shouldMarkMoveWhenFreePositionAvailable() {
+        //czy w przypadku tej metody wystarczy, że przetestuję jedno zaznaczenie position
+        // a może powinienem przetestować też i inne pola planszy?
+
+        //given
+        char[][] boardCopy = createBoard(EMPTY);
+        List<Position> positions = changeArrayToList(boardCopy);
+        Position position = new Position(0, 0);
+
+        //when
+        when(board.findFreePositions()).thenReturn(positions);
+        when(randomWrapper.nextInt(positions.size())).thenReturn(0);
+        computerPlayer.markMove(CIRCLE);
+
+        //then
+        assertEquals(9, board.findFreePositions().size());
+        verify(board).markMove(position, CIRCLE);
+    }
+
+    @Test
+    public void shouldNotMarkMoveWhenNoFreePositionsAvailable() {
```

Comment:
Moim zdaniem markPostion gdy nie da się oznaczyć pola, pwoinnien:
1. Rzucać excpetion - preferowane
albo
2. Zwracać status czy udało się oznaczyć pole

Ten test jest do przeniesienia do boardTest. To Board decyduje o tym czy oznaczy pole nie player!

Reply:
Rzuca exception

Reply:
Nie mam totalnie pojęcia jak miałoby wyglądać wtedy testowanie